distribution/sdk/sample-twitter-source/src/main/java/org/codice/ddf/catalog/twitter/source/TwitterSource.java

```diff
@@ -0,0 +1,269 @@
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.operation.impl.SourceResponseImpl;
+import ddf.catalog.resource.ResourceNotFoundException;
+import ddf.catalog.resource.ResourceNotSupportedException;
+import ddf.catalog.resource.ResourceReader;
+import ddf.catalog.source.FederatedSource;
+import ddf.catalog.source.SourceMonitor;
+import ddf.catalog.source.UnsupportedQueryException;
+import twitter4j.GeoLocation;
+import twitter4j.Query;
+import twitter4j.QueryResult;
+import twitter4j.Status;
+import twitter4j.Twitter;
+import twitter4j.TwitterException;
+import twitter4j.TwitterFactory;
+import twitter4j.conf.Configuration;
+import twitter4j.conf.ConfigurationBuilder;
+
+public class TwitterSource implements FederatedSource {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TwitterSource.class);
+
+    TwitterFactory twitterFactory;
+
+    String id;
+
+    ResourceReader resourceReader;
+
+    String consumerKey;
+
+    String consumerSecret;
+
+    public TwitterSource() {
+
+    }
+
+    public void init() {
+        ConfigurationBuilder configurationBuilder = new ConfigurationBuilder();
+        Configuration configuration = configurationBuilder.setOAuthConsumerKey(consumerKey)
+                .setOAuthConsumerSecret(consumerSecret)
+                .setApplicationOnlyAuthEnabled(true)
+                .build();
+
+        twitterFactory = new TwitterFactory(configuration);
+    }
+
+    public void destroy() {
+        twitterFactory = null;
+    }
+
+    @Override
+    public ResourceResponse retrieveResource(URI uri, Map<String, Serializable> arguments)
+            throws IOException, ResourceNotFoundException, ResourceNotSupportedException {
+        return null;
+    }
+
+    @Override
+    public Set<String> getSupportedSchemes() {
+        return Collections.emptySet();
+    }
+
+    @Override
+    public Set<String> getOptions(Metacard metacard) {
+        return Collections.emptySet();
+    }
+
+    @Override
+    public boolean isAvailable() {
+        Twitter instance = twitterFactory.getInstance();
+        try {
+            instance.getOAuth2Token();
+            return true;
+        } catch (TwitterException e) {
+            LOGGER.error("Unable to get OAuth2 token.", e);
+            return false;
+        }
+    }
+
+    @Override
+    public boolean isAvailable(SourceMonitor callback) {
+        if (isAvailable()) {
+            callback.setAvailable();
+            return true;
+        } else {
+            callback.setUnavailable();
+            return false;
+        }
+    }
+
+    @Override
+    public SourceResponse query(QueryRequest request) throws UnsupportedQueryException {
+        Twitter instance = twitterFactory.getInstance();
+        try {
+            instance.getOAuth2Token();
+        } catch (TwitterException e) {
+            LOGGER.error("Unable to get OAuth2 token.", e);
```

Comment:
Can you even execute the search below if you fail to get the OAuth token? Shouldn't this fail here?